db/scripts/compile_stats.py

```diff
@@ -0,0 +1,196 @@
+import MySQLdb
+import configparser
+import sys
+import json
+import logging, logging.config
+
+from datetime import datetime, timedelta, date
+from time import time
+
+import schema
+import queries
+from current_data import cur_states as states, get_elections
+
+config = configparser.ConfigParser()
+if not config.read('config.ini'):
+    raise Exception('config.ini not in current directory. Please run again from top-level directory.')
+
+logging.config.fileConfig(fname='log_config.ini')
+
+logger = logging.getLogger('dev')
+
+'''
+State-level statistics
+'''
+for state in states:
+    mydb = MySQLdb.connect(
+        host=config['DATABASE']['host'],
+        user=config['DATABASE']['user'],
+        passwd=config['DATABASE']['passwd'],
+        db=config[state]['db'],
+        local_infile=1)
+ 
+    cursor = mydb.cursor(MySQLdb.cursors.DictCursor)
+
+    cursor.execute(schema.create_state_stats_table())
+    
+    elections = get_elections(cursor)
+  
+    for election in elections:
+        logger.info(f'Computing {state} state-level statistics for {election}.')
+        t0 = time()
+        
+        cured_table = f'cured_{election}'
+        rej_table = f'rejected_{election}'
+
+        cursor.execute(queries.get_processed_count(election))
+        processed_tot = cursor.fetchall()[0]['num_processed']
+
+        cursor.execute(queries.get_cured_count(election))
+        cured_tot = cursor.fetchall()[0]['num_cured']
+
+        cursor.execute(queries.get_rej_count(election))
+        rej_tot = cursor.fetchall()[0]['num_rejected']
+
+        logger.debug(f'proc, cured, rej done @ {time() - t0:.2f}s')
+
+        cursor.execute(queries.get_rej_reasons(election))
+        rej_reasons = json.dumps(str(cursor.fetchall()))
```

Comment:
Small question about this. Why is it necessary to convert cursor.fetchall() result to a string since it should return a dictionary since the method using the dictCursor and json.dumps works on any python object right?

Reply:
Right - `json.dumps(str(cursor.fetchall()))` works if all the fields in the `cursor.fetchall()` are strings. However, in the case of integer/decimal numbers as in the case of the age group query, the json conversion to string fails unless we do something like this https://stackoverflow.com/questions/1960516/python-json-serialize-a-decimal-object. 

I worked around this by converting the whole thing to a string, which is why you saw the Decimal stuff in the age count column (for that - I cast it to a string in `queries.py` to remove the Decimal there as I mentioned earlier). I think my reasoning for doing these all as strings is in the case of another state having a similar issue, this would be a catch-all solution. 

Maybe we should just do the encoder and avoid doing string casting? It might make the string processing/casting easier on the stats endpoint. I think ideally, the stats endpoint should do everything through casting if possible, but string processing probably suffices for now.